client/src/routes/widgetModule/WidgetModule.tsx

```diff
@@ -95,6 +125,188 @@ export const WidgetModule: React.FC<Props> = observer(({ domainName }) => {
+        const setResult = await rootStore.easClient.setPublicAliases({
+          domainName,
+          aliases: newPublicAliases,
+          onTransactionHash: () => {
+            setProcessStatus({
+              type: ProcessStatusTypes.PROGRESS,
+              render: <BaseText>Waiting for transaction confirmation</BaseText>,
+            })
+          },
+        })
+
+        if (setResult.error) {
+          throw setResult.error
+        }
+
+        const delResult = await rootStore.easClient.deactivate({
+          domainName,
+          alias: removingAlias,
+          onTransactionHash: () => {
+            setProcessStatus({
+              type: ProcessStatusTypes.PROGRESS,
+              render: <BaseText>Waiting for transaction confirmation</BaseText>,
+            })
+          },
+        })
+
+        if (delResult.error) {
+          throw delResult.error
+        }
+
+        const { success, error } = await easServerClient.deactivate(
+          domainName,
+          removingAlias
+        )
+
+        if (error) {
+          throw new Error(error)
+        }
+      }
+
+      const signature = await rootStore.easClient.buildSignature(
+        domainName,
+        alias,
+        forward
+      )
+      const separator = ethers.utils.toUtf8Bytes(
+        await rootStore.easClient.getSeparator()
+      )
+      const data = ethers.utils.concat([
+        ethers.utils.toUtf8Bytes(alias),
+        separator,
+        ethers.utils.toUtf8Bytes(forward),
+        separator,
+        signature,
+      ])
+      let makePublic = true
+      const commitment = ethers.utils.keccak256(data)
+
+      console.log('### publicAliases', publicAliases)
+
+      if (publicAliases.includes(alias)) {
+        makePublic = false
+      }
+
+      setProcessStatus({
+        type: ProcessStatusTypes.PROGRESS,
+        render: (
+          <BaseText>
+            {walletStore.isMetamaskAvailable
+              ? 'Waiting for a transaction to be signed'
+              : 'Sign transaction on mobile device'}
+          </BaseText>
+        ),
+      })
+
+      const activateResult = await rootStore.easClient.activate({
+        domainName: domainName,
+        onTransactionHash: () => {
+          setProcessStatus({
+            type: ProcessStatusTypes.PROGRESS,
+            render: <BaseText>Waiting for transaction confirmation</BaseText>,
+          })
+        },
+        alias,
+        commitment,
+        makePublic,
+      })
+
+      if (activateResult.error) {
+        const message = activateResult.error.reason
+          ? activateResult.error.reason
+          : activateResult.error.message
+        setProcessStatus({
+          type: ProcessStatusTypes.ERROR,
+          render: <BaseText>{message}</BaseText>,
+        })
+        return
+      }
+
+      setProcessStatus({
+        type: ProcessStatusTypes.PROGRESS,
+        render: 'Adding email alias',
+      })
+
+      const { success, error } = await easServerClient.activate(
```

Comment:
similarly here; if this fails, the user would have an activated on-chain alias, but doesn't really work (not activated on server). Because the client only checks on-chain data, in the next attempt the client would try to deactivate the alias (assuming maxNumAlias=1) but would continue failing to do so (server deactivation call would return an error)